code/datums/callback.dm

```diff
@@ -54,37 +56,48 @@
 	if (length(args) > 2)
 		arguments = args.Copy(3)
 
+/proc/ImmediateInvokeAsync(thingtocall, proctocall, ...)
+	set waitfor = FALSE
 
-/datum/callback/proc/Invoke(...)
-	if (!object)
+	if (!thingtocall)
 		return
 
-	var/list/calling_arguments = arguments
+	var/list/calling_arguments = length(args) > 2 ? args.Copy(3) : null
+
+	if (thingtocall == GLOBAL_PROC)
+		call(proctocall)(arglist(calling_arguments))
+	else
+		call(thingtocall, proctocall)(arglist(calling_arguments))
+
+#define CALLBACK_INVOKE \
+	if (!object)\
+	{\
+		return;\
+	}\
+	var/list/calling_arguments = arguments;\
+	if (length(args))\
+	{\
+		if (length(arguments))\
+		{\
+			calling_arguments = calling_arguments + args; /*not += so that it creates a new list so the arguments list stays clean*/\
+		}\
+		else\
+		{\
+			calling_arguments = args;\
+		}\
```

Comment:
There's no need to be concerned about screen real estate, for the same reason we don't allow Coiax's 80 char lines, this isn't the 1900s.

Allman style is one of the most popular indent styles there are, fuck KR.